pilot/pkg/credentials/kube/secrets.go

```diff
@@ -80,15 +79,12 @@ func NewCredentialsController(kc kube.Client) *CredentialsController {
 	// This makes the assumption we will never care about Helm secrets or SA token secrets - two common
 	// large secrets in clusters.
 	// This is a best effort optimization only; the code would behave correctly if we watched all secrets.
-	fieldSelector := fields.AndSelectors(
-		fields.OneTermNotEqualSelector("type", "helm.sh/release.v1"),
-		fields.OneTermNotEqualSelector("type", string(v1.SecretTypeServiceAccountToken))).String()
-	secrets := kclient.NewFiltered[*v1.Secret](kc, kclient.Filter{
-		FieldSelector: fieldSelector,
-	})
-
+	// TODO: Shouldn't we use "github.com/maistra/xns-informer/pkg/generated/kube/core/v1".NewFilteredSecretInformer with
+	// fieldSelector := fields.AndSelectors(
+	//     fields.OneTermNotEqualSelector("type", "helm.sh/release.v1"),
+	//     fields.OneTermNotEqualSelector("type", string(v1.SecretTypeServiceAccountToken))).String()
```

Comment:
It's not filtered in previous releases, so I wasn't sure if I should apply this change in rebase, because we may need additional tests... What do you think?

Reply:
`xns-informers` watches resources from the namespace set, not in all namespaces. If we used upstream informer, we would get permission errors like [here](https://github.com/maistra/istio/pull/803).

Reply:
Anyway, I will investigate it later.